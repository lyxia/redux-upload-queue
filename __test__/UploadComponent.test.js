import config, {store, Foo} from './UploadConfig'
import uploadComponent from '../UploadComponent'

import React from 'react'
import { Provider } from 'react-redux'
import { mount } from 'enzyme';

afterEach(() => {
    store.clearActions()
    fetch.resetMocks()
})

test('uploadComponent new', () => {
    fetch.mockResponseOnce(JSON.stringify({ error: null, id: '123456' }))
    
    const Component = uploadComponent({ upload: 'uploadKey', config: config })(Foo)
    const componentWrap = mount(
        <Provider store={store}>
            <Component />
        </Provider>
    )
    const fooWrap = componentWrap.find(Foo)
    const fooProps = fooWrap.props()
    fooProps.pushUploadItem('fileOnePath', 'fileOne')
    return fooProps.startUpload().then(() => {
        expect(store.getActions()).toMatchSnapshot()
    })
})

test('uploadComponent old', () => {
    fetch.mockResponseOnce(JSON.stringify({ error: null, id: '123456' }))
    
    const Component = uploadComponent({
        upload: 'uploadKey',
        uploadApi: config.uploadApi, //api.uploadImageWithPost
        getUUIDFuc: config.getUUIDFuc,
        isSuccess: config.isSuccess,
    })(Foo)
    const componentWrap = mount(
        <Provider store={store}>
            <Component />
        </Provider>
    )
    const fooWrap = componentWrap.find(Foo)
    const fooProps = fooWrap.props()
    fooProps.pushUploadItem('fileOnePath', 'fileOne')
    return fooProps.startUpload().then(() => {
        expect(store.getActions()).toMatchSnapshot()
    })
})

