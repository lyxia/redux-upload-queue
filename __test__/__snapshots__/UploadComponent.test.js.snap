// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uploadComponent new 1`] = `
Array [
  Object {
    "payload": Object {
      "upload": "uploadKey",
    },
    "type": "Upload/registerUpload",
  },
  Object {
    "payload": Object {
      "filePath": "fileOnePath",
      "name": "fileOne",
      "upload": "uploadKey",
    },
    "type": "Upload/pushItem",
  },
  Object {
    "payload": Object {
      "upload": "uploadKey",
    },
    "type": "Upload/startUpload",
  },
  Object {
    "payload": Object {
      "upload": "uploadKey",
    },
    "type": "Upload/startUploadItem",
  },
  Object {
    "payload": Object {
      "id": "123456",
      "upload": "uploadKey",
    },
    "type": "Upload/successUploadItem",
  },
  Object {
    "payload": Object {
      "upload": "uploadKey",
    },
    "type": "Upload/uploadComplete",
  },
]
`;

exports[`uploadComponent old 1`] = `
Array [
  Object {
    "payload": Object {
      "upload": "uploadKey",
    },
    "type": "Upload/registerUpload",
  },
  Object {
    "payload": Object {
      "filePath": "fileOnePath",
      "name": "fileOne",
      "upload": "uploadKey",
    },
    "type": "Upload/pushItem",
  },
  Object {
    "payload": Object {
      "upload": "uploadKey",
    },
    "type": "Upload/startUpload",
  },
  Object {
    "payload": Object {
      "upload": "uploadKey",
    },
    "type": "Upload/startUploadItem",
  },
  Object {
    "payload": Object {
      "id": "123456",
      "upload": "uploadKey",
    },
    "type": "Upload/successUploadItem",
  },
  Object {
    "payload": Object {
      "upload": "uploadKey",
    },
    "type": "Upload/uploadComplete",
  },
]
`;
